tla-types/src/main/scala/at/forsyte/apalache/tla/typecheck/etc/EqClass.scala

```diff
@@ -18,15 +18,43 @@ class EqClass {
   /**
    * Type variables that are associated with the equivalence class.
    */
-  var typeVars: Set[Int] = Set.empty
+  private var _typeVars: Set[Int] = Set.empty
+
+  /**
+   * The representative of an equivalence class.
+   */
+  private var _reprVar: Int = -1
```

Comment:
Could this either be `Option[Int]` have the initial value set via an an argument to the class constructor? It seems to me that either would be preferable to having an invalid value for initialization.

Reply:
Refactored `EqClass` too. This will also change in #973.